File: src/main/java/pl/ptroc/hotel/springhotelreservation/model/Booking.java
package pl.ptroc.hotel.springhotelreservation.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

/**
 * Created by Paweł Troć on 2018-01-06.
 */
@Entity
@Table(name = "booking")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Booking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "hotel_room_id")
    private HotelRoom hotelRoom;

/*    @Column(name = "customer_id")
    private Customer customer;*/

    @Column(name = "begin_date")
    private LocalDate beginDate;

    @Column(name = "end_date")
    private LocalDate endDate;
}

File: angular-app/src/app/hotel-room/Booking.ts
import { Room } from './Room';


export class Booking {
	room: Room;
	beginDate: Date;
	endDate: Date;
	
	constructor(room: Room, beginDate: Date, endDate: Date) {
		this.room = room;
		this.beginDate = beginDate;
		this.endDate = endDate; 
	}
}